public/app/plugins/panel/dashlist/models.cue

```diff
@@ -0,0 +1,21 @@
+package grafanaschema
+
+Family: {
+    lineages: [
+        [
+            {
+                PanelOptions: {
+                    showStarred: bool | *true
+                    showRecentlyViewed: bool | *false
+                    showSearch: bool | *false
+                    showHeadings: bool | *true
+                    maxItems: int | *10;
+                    query: string | *""
+                    folderId?: int
+                    tags: string[] | *[]
```

Comment:
Hello @ryantxu - sorry still stuck on CI stuff. With a first look it seems that these changes are needed IIUC? 
```suggestion
                    maxItems: int | *10
                    query: string | *""
                    folderId?: int
                    tags: [...string] | *[]
```
then there's `required field PanelFieldConfig is optional in subsumed value` which is another thing I guess - maybe something related to what Ying's doing?

Reply:
Yup, this change is correct. `[...string]` is how you express "a list of at least zero length with no maximum length, where all elements are a string".

This error: `required field PanelFieldConfig is optional in subsumed value` is a pretty classic example of a poor CUE error, and one we'll plausibly want to wrap.

It's saying that the schema being declared in this file - the object created starting on line 6 - is missing the required field `PanelFieldConfig`. That requirement is dictated by the `#PanelSchema` definition, [here](https://github.com/grafana/grafana/blob/master/cue/scuemata/panel-plugin.cue#L11), as we expect that all families in these files are instances of [`#PanelFamily`](https://github.com/grafana/grafana/blob/master/cue/scuemata/panel-plugin.cue#L18). Subsumption is mentioned because that's the [method](https://pkg.go.dev/cuelang.org/go@v0.4.0-alpha.1/cue#Value.Subsume) relied upon to check if the schema declared in this models.cue file is an instance of that `#PanelSchema`.

Reply:
More concrete, for absolute clarity. The error `required field PanelFieldConfig is optional in subsumed value` happens because this structure in this file:

```cue
            {
                PanelOptions: {
                    showStarred: bool | *true
                    showRecentlyViewed: bool | *false
                    showSearch: bool | *false
                    showHeadings: bool | *true
                    maxItems: int | *10;
                    query: string | *""
                    folderId?: int
                    tags: string[] | *[]
                }
            }
```

Does not contain a `PanelFieldConfig` field, which is specified as required by [`#PanelSchema`](https://github.com/grafana/grafana/blob/0d1cdbe227fe56077559f23f67c1a3f6785709eb/cue/scuemata/panel-plugin.cue#L9-L12):
```cue
#PanelSchema: {
    PanelOptions: {...}
    PanelFieldConfig: {...}
}
```

There are a few possible paths to addressing this, not all mutually exclusive:

* Improve the error message, as i noted in my previous comment.
* Make `PanelFieldConfig` an optional field in `#PanelSchema`.
* Expect that `models.cue` files import and unify the `#PanelFamily`; i.e., line 3 of this file becomes `Family: #PanelFamily & {`. Benefit might be that the file is more explicit about what's happening to the reader (assuming they understand CUE); cost is that it seems like a magical incantation to the file's author if they're inexperienced with CUE.
* Do the same unification as in the previous bullet, but do it on the fly in Go code. It'd have to be before [this call](https://github.com/grafana/grafana/blob/1336a57e99788e267e1ff882a33b005d66ad8b21/pkg/schema/load/panel.go#L142), which is where the error arises. More magical, but less confusing for folks inexperienced with CUE.

Reply:
@ryantxu, @sdboyer just saw that Ryan made the field optional - is this the way to go in the end?

Reply:
yes -- most panels do not need to define custom field config.